base/applications/notepad/text.c

```diff
@@ -47,6 +48,32 @@ static BOOL Append(LPWSTR *ppszText, DWORD *pdwTextLen, LPCWSTR pszAppendText, D
     return TRUE;
 }
 
+ENCODING AnalyzeEncoding(const char *pBytes, DWORD dwSize)
+{
+    INT flags = IS_TEXT_UNICODE_STATISTICS;
+
+    if (dwSize <= 1)
+        return ENCODING_ANSI;
+
+    if (IsTextUnicode(pBytes, dwSize, &flags))
+    {
+        return ENCODING_UTF16LE;
+    }
+
+    if ((flags & IS_TEXT_UNICODE_REVERSE_MASK) && !(flags & IS_TEXT_UNICODE_ILLEGAL_CHARS))
```

Comment:
Do you have a big endian file that you're testing with? Intuitively it would seem like `IsTextUnicode` should also return TRUE for BE strings.

Reply:
```c
#include <windows.h>
#include <stdio.h>

int main(void)
{
    INT flags = IS_TEXT_UNICODE_STATISTICS;
    if (IsTextUnicode("\x00\n", 2, &flags))
    {
        puts("OK");
    }
    else
    {
        puts("NG");
    }
    return 0;
}
```
The result is NG.